src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/EditProjectFileCommand.cs

```diff
@@ -60,31 +74,36 @@ internal abstract class AbstractEditProjectFileCommand : AbstractSingleNodeProje
         {
             if (!ShouldHandle(node)) return false;
 
-            var projectFileName = Path.GetFileName(_unconfiguredProject.FullPath);
-
-            var projPath = await GetFileAsync(projectFileName).ConfigureAwait(false);
             await _threadingService.SwitchToUIThread();
-            IVsWindowFrame frame;
 
-            frame = _shellUtilities.OpenDocumentWithSpecificEditor(_serviceProvider, projPath, XmlEditorFactoryGuid, Guid.Empty);
+            // If the window has already been opened and hasn't yet been closed, reshow it.
+            if (_isInitialized)
+            {
+                _frame.Show();
+                return true;
+            }
+
+            await SetupFileAsync().ConfigureAwait(true);
 
             IMsBuildModelWatcher watcher = _watcherFactory.CreateExport();
-            await watcher.InitializeAsync(projPath).ConfigureAwait(true);
+            await watcher.InitializeAsync(_tempProjectPath, _lastWrittenXml).ConfigureAwait(true);
+
+            _frame = _shellUtilities.OpenDocumentWithSpecificEditor(_serviceProvider, _tempProjectPath, XmlEditorFactoryGuid, Guid.Empty);
 
             // When the document is closed, clean up the file on disk
-            var fileCleanupListener = new EditProjectFileCleanupFrameNotifyListener(projPath, _fileSystem, watcher);
-            Verify.HResult(frame.SetProperty((int)__VSFPROPID.VSFPROPID_ViewHelper, fileCleanupListener));
+            var fileCleanupListener = new EditProjectFileVsFrameEvents(_tempProjectPath, _fileSystem, watcher, _frame, _serviceProviderHelper, this);
+            fileCleanupListener.InitializeEvents();
 
             // Ensure that the window is not reopened when the solution is closed
-            Verify.HResult(frame.SetProperty((int)__VSFPROPID5.VSFPROPID_DontAutoOpen, true));
+            Verify.HResult(_frame.SetProperty((int)__VSFPROPID5.VSFPROPID_DontAutoOpen, true));
 
```

Comment:
The temp file is cleaned up when we close the editor, and the logic to set up this editor isn't run when the editor is opened. The editor would break if opened again.